packages/workflow/src/worker-interface.ts

```diff
@@ -128,7 +129,7 @@ export async function initRuntime({ info, randomnessSeed, now, patches }: Workfl
   const customPayloadConverter = (await import('__temporal_custom_payload_converter')).payloadConverter;
   // The `payloadConverter` export is validated in the Worker
   if (customPayloadConverter !== undefined) {
-    state.payloadConverter = customPayloadConverter;
+    state.payloadConverter = new WrappedCustomPayloadConverter(customPayloadConverter);
```

Comment:
I missed this the first go around. Eventually found it through the `toPayload throwing results in task retry` test